app/src/main/java/com/andrew/tasky/presentation/task_detail/TaskDetailFragment.kt

```diff
@@ -4,29 +4,198 @@ import androidx.lifecycle.ViewModelProvider
+            navController.navigate(R.id.action_taskDetailFragment_to_editFragment, bundle)
+        }
+
+        binding.descriptionTextView.setOnClickListener {
+            val editType= EditType.DESCRIPTION.ordinal
+            val bundle = bundleOf("editType" to editType)
+            navController.navigate(R.id.action_taskDetailFragment_to_editFragment, bundle)
+        }
+
+        binding.time.setOnClickListener {
+            timePickerFragment()
+        }
+
+        binding.date.setOnClickListener {
+            datePickerFragment()
+        }
+    }
+
+    private fun subscribeToObservables() {
+        lifecycleScope.launchWhenStarted {
+            viewModel.selectedDate.collectLatest {
+                binding.date.text = it
+            }
+        }
+        lifecycleScope.launchWhenStarted {
+            viewModel.selectedTime.collectLatest {
+                binding.time.text = it
+            }
+        }
+    }
+
+    private fun timePickerFragment(){
+
+        val timePickerFragment = TimePickerFragment()
+        val supportFragmentManager = requireActivity().supportFragmentManager
+
+        supportFragmentManager.setFragmentResultListener(
+            "REQUEST_KEY",
+            viewLifecycleOwner
+        ){
+                resultKey, bundle -> if(resultKey == "REQUEST_KEY"){
+            val time = bundle.getString("SELECTED_TIME")
+            viewModel.setSelectedTime(time)
+        }
+        }
+        timePickerFragment.show(supportFragmentManager, "TimePickerFragment")
+    }
+
+    private fun datePickerFragment(){
+        val datePickerFragment = DatePickerFragment()
+        val supportFragmentManager = requireActivity().supportFragmentManager
+
+        supportFragmentManager.setFragmentResultListener(
+            "REQUEST_KEY",
+            viewLifecycleOwner
+        ){
+                resultKey, bundle -> if(resultKey == "REQUEST_KEY"){
+            val date = bundle.getString("SELECTED_DATE")
+            viewModel.setSelectedDate(date)
+        }
+        }
+        datePickerFragment.show(supportFragmentManager, "DatePickerFragment")
+    }
+
+    private fun editMode(){
+        binding.currentDateTextView.text = getString(R.string.edit_task_text)
+
+        binding.saveButton.isVisible = true
+        binding.saveButton.isClickable = true
+
+        binding.editButton.isVisible = false
+        binding.editButton.isClickable = false
+
+        binding.editTitleButton.isVisible = true
+        binding.editTitleButton.isClickable = true
+        binding.taskTitle.isClickable = true
+
+        binding.editDescriptionButton.isVisible = true
+        binding.editDescriptionButton.isClickable = true
+        binding.descriptionTextView.isClickable = true
+
+        binding.time.isClickable = true
+        binding.date.isClickable = true
+
+        binding.reminderButton.isClickable = true
+        binding.reminderButton.isVisible = true
+    }
+
+    private fun nonEditMode(){
```

Comment:
This function is redundant, you can instead create one like this:
```
private fun setEditMode(isEditing: Boolean){
        binding.currentDateTextView.text = getString(R.string.edit_task_text)

        binding.saveButton.isVisible = isEditing
        binding.saveButton.isClickable = isEditing

        binding.editButton.isVisible = !isEditing
        binding.editButton.isClickable = !isEditing

        binding.editTitleButton.isVisible = isEditing
        binding.editTitleButton.isClickable = isEditing
        binding.taskTitle.isClickable = isEditing

        binding.editDescriptionButton.isVisible = isEditing
        binding.editDescriptionButton.isClickable = isEditing
        binding.descriptionTextView.isClickable = isEditing

        binding.time.isClickable = isEditing
        binding.date.isClickable = isEditing

        binding.reminderButton.isClickable = isEditing
        binding.reminderButton.isVisible = isEditing
    }
```
and you use that passed boolean to update your views. Considering the feedback from above, you can then simply call this in the observer of your `isInEditMode´ Flow from the ViewModel